controllers/metrics/dashboard_controller.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"context"
+
+	log "github.com/ViaQ/logerr/v2/log/static"
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/k8s/loader"
+	"github.com/openshift/cluster-logging-operator/internal/metrics"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ReconcileDashboard struct {
+	Client client.Client
+
+	Reader client.Reader
+}
+
+func (r *ReconcileDashboard) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	log.V(3).Info("Reconciling collector dashboard")
+
+	// Only reconcile collector dashboard only if there are forwarders or a legacy clusterlogging instance
+	// Get list of CLFs
+	clfList := &loggingv1.ClusterLogForwarderList{}
+	if err := r.Client.List(context.TODO(), clfList); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Check if CL named instance is available
+	_, err := loader.FetchClusterLogging(r.Client, constants.OpenshiftNS, constants.SingletonName, true)
+
+	// Remove dashboard when no CLFs and no CL named instance
+	if len(clfList.Items) == 0 && (err != nil && errors.IsNotFound(err)) {
+		log.V(3).Info("No CLF or CLs, removing dashboard")
+		if err := metrics.RemoveDashboardConfigMap(r.Client); err != nil {
+			if !errors.IsNotFound(err) {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Reconcile dashboard
+	err = metrics.ReconcileDashboards(r.Client, r.Reader)
+
+	return ctrl.Result{}, err
+}
+
+func (r *ReconcileDashboard) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}).
```

Comment:
I've reframed the way the controller works. It will react to any CLF/CL events and reconcile a dashboard based on `len(CLFs) > 1` && `CL.Name == 'instance' && CL.Namespace == 'openshift-logging'`